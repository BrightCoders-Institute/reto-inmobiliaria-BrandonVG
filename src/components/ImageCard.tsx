import React from 'react';
import {ImageBackground, StyleSheet, Text, View} from 'react-native';
import Fontisto from 'react-native-vector-icons/Fontisto';

type imageCardProps = {
  img: string;
  stars: string;
};

const ImageCard = (props: imageCardProps) => {
  return (
    <ImageBackground source={{uri: props.img}} style={styles.img}>
      <View style={styles.starsContainer}>
        <Fontisto name="star" color="#EEBA00" size={11} />
        <Text style={styles.text}>{props.stars}</Text>
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  img: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    borderRadius: 10,
    overflow: 'hidden',
  },
  starsContainer: {
    flexDirection: 'row',
    backgroundColor: '#FBEDB7',
    justifyContent: 'center',
    alignItems: 'center',
    height: 25,
    width: 50,
    borderRadius: 20,
    margin: 5,
  },
  text: {
    color: '#856D35',
    marginLeft: 1,
    fontSize: 14,
    fontWeight: 'bold',
  },
});

export default ImageCard;
