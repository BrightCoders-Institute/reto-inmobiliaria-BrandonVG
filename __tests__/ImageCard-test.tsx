import React from 'react';
import {create} from 'react-test-renderer';
import ImageCard from '../src/components/ImageCard';

const tree = create(
  <ImageCard
    img="https://definicion.de/wp-content/uploads/2011/01/casa-2.jpg"
    stars="4.4"
  />,
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
