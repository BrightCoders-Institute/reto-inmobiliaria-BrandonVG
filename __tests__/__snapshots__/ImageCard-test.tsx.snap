// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "alignItems": "center",
      "borderRadius": 10,
      "flex": 1,
      "justifyContent": "flex-end",
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://definicion.de/wp-content/uploads/2011/01/casa-2.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FBEDB7",
        "borderRadius": 20,
        "flexDirection": "row",
        "height": 25,
        "justifyContent": "center",
        "margin": 5,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#EEBA00",
            "fontSize": 11,
          },
          undefined,
          {
            "fontFamily": "Fontisto",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î§©
    </Text>
    <Text
      style={
        {
          "color": "#856D35",
          "fontSize": 14,
          "fontWeight": "bold",
          "marginLeft": 1,
        }
      }
    >
      4.4
    </Text>
  </View>
</View>
`;
